<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>What the Package Does (One Line, Title Case) • gwangmyeongseong3</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="What the Package Does (One Line, Title Case)">
<meta property="og:description" content="What the package does (one paragraph).">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">gwangmyeongseong3</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9001</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/lodestar_start_guide.html">lodestar_start_guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="gwangmyeongseong3" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#gwangmyeongseong3" class="anchor"></a>gwangmyeongseong3</h1></div>
<p>This a redux of the Sputnik and OpenSputnik comparative genomics projects.</p>
<p>This is another vanity project aiming to rekindle some bioinformatics and coding passion.</p>
<div id="why-gwangmyeongseong3" class="section level2">
<h2 class="hasAnchor">
<a href="#why-gwangmyeongseong3" class="anchor"></a>Why gwangmyeongseong3?</h2>
<p>In the beginning there were Sanger-based DNA sequence reads produced from purified cDNA. A robust sequencing library could contain between 1000s and 100,000s of sequences that with some patience could be filtered, clustered and assembled into Unigene sets.</p>
<p>With the advent of short-read DNA sequencing it was simpler, faster and computationally more scary to perform such <em>ab initio</em> based investigation of the transcriptome. Third generation DNA sequencing such as Oxford Nanopore’s <a href="https://store.nanoporetech.com/eu/catalog/product/view/id/297/s/direct-rna-sequencing-kit/category/28/">direct RNA sequencing</a> or <a href="https://store.nanoporetech.com/eu/sample-prep/direct-cdna-sequencing-kit.html">cDNA sequencing</a> offers the ability to sequence full-length transcripts and to fully characterise the diversity of transcripts and their isoforms.</p>
<p>Earlier bioinformatics solutions such as openSputnik (4) provided schedulers for the distributed bioinformatics required for the annotation and analysis of sequence collections - some of these solutions have been lost to time. There is certainly an <a href="https://en.wikipedia.org/wiki/List_of_RNA-Seq_bioinformatics_tools">over-abundance of tools</a> for analysing RNA-Seq data - nothing really jumps out as a solution for the exploration of full-length cDNA sequence data in the absence of reference genome sequences … this is where we are starting.</p>
<p>With the development of fabulous workflow software such as <a href="https://snakemake.readthedocs.io/en/stable/">Snakemake</a> and <a href="https://www.nextflow.io/">Nextflow</a> some of the scheduling requirements are already very much more capable than workflows provided in the past.</p>
</div>
<div id="ambitions-for-gwangmyeongseong3" class="section level2">
<h2 class="hasAnchor">
<a href="#ambitions-for-gwangmyeongseong3" class="anchor"></a>Ambitions for gwangmyeongseong3?</h2>
<ul>
<li>To facilitate the ab initio annotation and analysis of full-length cDNA sequence data</li>
<li>To leverage existing workflow / pipeline software</li>
<li>To re-utilise RDBMS software for the structured storage and curation of biological data</li>
<li>To provide coherent and dynamic reporting of the experimental and annotative data</li>
<li>To be completely open and reproducible</li>
<li>To utilise Tidyverse compliant R in so far as is possible throughout</li>
<li>To (finally) containerise the software</li>
<li>Provision of meaningful documentation (pkgdown)</li>
</ul>
</div>
<div id="history---from-sputnik-to-opensputnik" class="section level2">
<h2 class="hasAnchor">
<a href="#history---from-sputnik-to-opensputnik" class="anchor"></a>History - from Sputnik to openSputnik</h2>
<p>As a postdoc at the turn of the millenium, I was tasked with adapting the PEDANT software (1) for usage within comparative plant genomics. At the time we had a single plant genome available (2) so the plan was to focus on strategies for the clustering, assembly and annotation of plant EST sequences. The Sputnik (3) project was born in 2001 - it was implemented in Python and used a PostgreSQL database at the backend to structure data and enable intelligent queries of salient information.</p>
<p>In 2004 I moved to Finland to start my own research group. Sputnik couldn’t travel with me so a new project was born, OpenSputnik (4). This was a complete reimplementation of the original concepts borrowed from PEDANT and offered a simpler (although still beastly) implementation based entirely in JAVA. The software was intended to be open and there were contributions to <a href="https://sourceforge.net/projects/opensputnik/files/">SourceForge</a> - I am not aware of the source code being available anywhere or still existing - it has all been lost to time.</p>
<p>The OpenSputnik platform was used in a load of collaborations in a variery of different species but I was recruited into the pharmaceutical industry in 2006 and my ambitions to further develop and maintain the software were clipped.</p>
<p>A couple of collaborators have enquired about the persistence of their datasets and availability of software in the 1.5 decades since …</p>
</div>
<div id="redux-and-kwangmyŏngsŏng-3-unit-2" class="section level2">
<h2 class="hasAnchor">
<a href="#redux-and-kwangmy%C5%8Fngs%C5%8Fng-3-unit-2" class="anchor"></a>Redux and Kwangmyŏngsŏng-3 Unit 2</h2>
<p>The original Sputnik name was derived from the <a href="https://en.wikipedia.org/wiki/Sputnik_Sweetheart">title of a book</a> during a brainstorm with my mentor, Klaus FX Mayer. Although a very much better name for a microsatellite detection software (5), the Sputnik name is used in more recent bioinformatics software (in R noless) for peak analysis (6) and bioinformatics distributed computing (7).</p>
<p>We need a new name. I really like the <code>Lodestar</code> naming of Korean satellites but this <a href="http://ebispot.github.io/lodestar/lodestar.html">Lodestar nomenclature</a> is already used by the EBI. I have thus chosen to go with the Korean word for Lodestar. This is as already suggested a vanity project - in the event that this project is actually used by anyone then apologies for the excessive <code>gwangmyeongseong3</code> name.</p>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<ol>
<li><p>Riley, M. L., Schmidt, T., Artamonova, I. I., Wagner, C., Volz, A., Heumann, K., Mewes, H. W., &amp; Frishman, D. (2007). PEDANT genome database: 10 years online. Nucleic acids research, 35(Database issue), D354–D357. <a href="https://doi.org/10.1093/nar/gkl1005" class="uri">https://doi.org/10.1093/nar/gkl1005</a></p></li>
<li><p>Arabidopsis Genome Initiative (2000). Analysis of the genome sequence of the flowering plant Arabidopsis thaliana. Nature, 408(6814), 796–815. <a href="https://doi.org/10.1038/35048692" class="uri">https://doi.org/10.1038/35048692</a></p></li>
<li><p>Rudd, S., Mewes, H. W., &amp; Mayer, K. F. (2003). Sputnik: a database platform for comparative plant genomics. Nucleic acids research, 31(1), 128–132. <a href="https://doi.org/10.1093/nar/gkg075" class="uri">https://doi.org/10.1093/nar/gkg075</a></p></li>
<li><p>Rudd S. (2005). openSputnik–a database to ESTablish comparative plant genomics using unsaturated sequence collections. Nucleic acids research, 33(Database issue), D622–D627. <a href="https://doi.org/10.1093/nar/gki040" class="uri">https://doi.org/10.1093/nar/gki040</a></p></li>
<li><p>Robinson, A. J., Love, C. G., Batley, J., Barker, G., &amp; Edwards, D. (2004). Simple sequence repeat marker loci discovery using SSR primer. Bioinformatics (Oxford, England), 20(9), 1475–1476. <a href="https://doi.org/10.1093/bioinformatics/bth104" class="uri">https://doi.org/10.1093/bioinformatics/bth104</a></p></li>
<li><p>Inglese, P., Correia, G., Takats, Z., Nicholson, J. K., &amp; Glen, R. C. (2019). SPUTNIK: an R package for filtering of spatially related peaks in mass spectrometry imaging data. Bioinformatics (Oxford, England), 35(1), 178–180. <a href="https://doi.org/10.1093/bioinformatics/bty622" class="uri">https://doi.org/10.1093/bioinformatics/bty622</a></p></li>
<li><p>Völkel, G., Lausser, L., Schmid, F., Kraus, J. M., &amp; Kestler, H. A. (2015). Sputnik: ad hoc distributed computation. Bioinformatics (Oxford, England), 31(8), 1298–1301. <a href="https://doi.org/10.1093/bioinformatics/btu818" class="uri">https://doi.org/10.1093/bioinformatics/btu818</a></p></li>
</ol>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>`use_mit_license()`, `use_gpl3_license()` or friends to
    pick a license</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Stephen Rudd <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-0344-7487" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Stephen Rudd.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
